homeassistant/components/yardian/coordinator.py

```diff
@@ -0,0 +1,78 @@
+"""Update coordinators for Yardian."""
+
+from __future__ import annotations
+
+import asyncio
+from dataclasses import dataclass
+import datetime
+import logging
+
+from pyyardian import AsyncYardianClient, NetworkException, NotAuthorizedException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, MANUFACTURER
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = datetime.timedelta(seconds=30)
+
+
+@dataclass
+class YardianDeviceState:
+    """Data retrieved from a Yardian device."""
+
+    zones: list[list]
+    active_zones: set[int]
+
+
+class YardianUpdateCoordinator(DataUpdateCoordinator[YardianDeviceState]):
+    """Coordinator for Yardian API calls."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry: ConfigEntry,
+        controller: AsyncYardianClient,
+    ) -> None:
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=entry.title,
+            update_method=self._async_update_data,
+            update_interval=SCAN_INTERVAL,
+        )
+
+        self.controller = controller
+        self.yid = entry.data["yid"]
+        self._name = entry.title
+        self._model = entry.data["model"]
+        self._amount_of_zones = entry.data["zones"]
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return information about the device."""
+        return DeviceInfo(
+            name=self._name,
+            identifiers={(DOMAIN, self.yid)},
+            manufacturer=MANUFACTURER,
+            model=self._model,
+        )
+
+    async def _async_update_data(self) -> YardianDeviceState:
+        """Fetch data from Yardian device."""
+        try:
+            async with asyncio.timeout(10):
+                zones = await self.controller.fetch_zone_info(self._amount_of_zones)
```

Comment:
Not required, but it could make futures reviews smaller: Since you are wrapping the data, you could move these lines into the library and return the data object directly

Reply:
Ah yes. It makes sense. I think I'll do it in this PR to make codes here cleaner